// @wordpress/components/build/angle-picker-control/angle-circle.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// packages/components/src/angle-picker-control/angle-circle.tsx
var angle_circle_exports = {};
__export(angle_circle_exports, {
  default: () => angle_circle_default
});
module.exports = __toCommonJS(angle_circle_exports);
var import_clsx = __toESM(require("clsx"));
var import_element = require("@wordpress/element");
var import_compose = require("@wordpress/compose");

// packages/components/src/angle-picker-control/style.module.scss
var css = `/**
 * SCSS Variables.
 *
 * Please use variables from this sheet to ensure consistency across the UI.
 * Don't add to this sheet unless you're pretty sure the value will be reused in many places.
 * For example, don't add rules to this sheet that affect block visuals. It's purely for UI.
 */
/**
 * Colors
 */
/**
 * Fonts & basic variables.
 */
/**
 * Typography
 */
/**
 * Grid System.
 * https://make.wordpress.org/design/2019/10/31/proposal-a-consistent-spacing-system-for-wordpress/
 */
/**
 * Radius scale.
 */
/**
 * Elevation scale.
 */
/**
 * Dimensions.
 */
/**
 * Mobile specific styles
 */
/**
 * Editor styles.
 */
/**
 * Block & Editor UI.
 */
/**
 * Block paddings.
 */
/**
 * React Native specific.
 * These variables do not appear to be used anywhere else.
 */
.style-module__circle-root__j1e41 {
  border-radius: 50%;
  border: 1px solid var(--wp-components-color-gray-600, #949494);
  box-sizing: border-box;
  cursor: grab;
  height: 32px;
  overflow: hidden;
  width: 32px;
}
.style-module__circle-root__j1e41:active {
  cursor: grabbing;
}

.style-module__circle-indicator-wrapper__sbrph {
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
}
.style-module__circle-indicator-wrapper__sbrph:focus-visible {
  outline: none;
}

.style-module__circle-indicator__bS-go {
  background: var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  left: 50%;
  top: 4px;
  transform: translateX(-50%);
  position: absolute;
  width: 6px;
  height: 6px;
}`;
document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(css));
var style_module_default = {
  "circle-root": "style-module__circle-root__j1e41",
  "circle-indicator-wrapper": "style-module__circle-indicator-wrapper__sbrph",
  "circle-indicator": "style-module__circle-indicator__bS-go"
};

// packages/components/src/angle-picker-control/angle-circle.tsx
var import_jsx_runtime = require("react/jsx-runtime");
function AngleCircle({
  value,
  onChange,
  className,
  ...props
}) {
  const angleCircleRef = (0, import_element.useRef)(null);
  const angleCircleCenterRef = (0, import_element.useRef)();
  const previousCursorValueRef = (0, import_element.useRef)();
  const setAngleCircleCenter = () => {
    if (angleCircleRef.current === null) {
      return;
    }
    const rect = angleCircleRef.current.getBoundingClientRect();
    angleCircleCenterRef.current = {
      x: rect.x + rect.width / 2,
      y: rect.y + rect.height / 2
    };
  };
  const changeAngleToPosition = (event) => {
    if (event === void 0) {
      return;
    }
    event.preventDefault();
    event.target?.focus();
    if (angleCircleCenterRef.current !== void 0 && onChange !== void 0) {
      const {
        x: centerX,
        y: centerY
      } = angleCircleCenterRef.current;
      onChange(getAngle(centerX, centerY, event.clientX, event.clientY));
    }
  };
  const {
    startDrag,
    isDragging
  } = (0, import_compose.__experimentalUseDragging)({
    onDragStart: (event) => {
      setAngleCircleCenter();
      changeAngleToPosition(event);
    },
    onDragMove: changeAngleToPosition,
    onDragEnd: changeAngleToPosition
  });
  (0, import_element.useEffect)(() => {
    if (isDragging) {
      if (previousCursorValueRef.current === void 0) {
        previousCursorValueRef.current = document.body.style.cursor;
      }
      document.body.style.cursor = "grabbing";
    } else {
      document.body.style.cursor = previousCursorValueRef.current || "";
      previousCursorValueRef.current = void 0;
    }
  }, [isDragging]);
  return (
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", {
      ref: angleCircleRef,
      onMouseDown: startDrag,
      className: (0, import_clsx.default)("components-angle-picker-control__angle-circle", style_module_default["circle-root"], className),
      ...props,
      children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", {
        style: value ? {
          transform: `rotate(${value}deg)`
        } : void 0,
        className: (0, import_clsx.default)("components-angle-picker-control__angle-circle-indicator-wrapper", style_module_default["circle-indicator-wrapper"]),
        tabIndex: -1,
        children: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", {
          className: (0, import_clsx.default)("components-angle-picker-control__angle-circle-indicator", style_module_default["circle-indicator"])
        })
      })
    })
  );
}
function getAngle(centerX, centerY, pointX, pointY) {
  const y = pointY - centerY;
  const x = pointX - centerX;
  const angleInRadians = Math.atan2(y, x);
  const angleInDeg = Math.round(angleInRadians * (180 / Math.PI)) + 90;
  if (angleInDeg < 0) {
    return 360 + angleInDeg;
  }
  return angleInDeg;
}
var angle_circle_default = AngleCircle;
