// @wordpress/components/build/alignment-matrix-control/cell.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// packages/components/src/alignment-matrix-control/cell.tsx
var cell_exports = {};
__export(cell_exports, {
  default: () => Cell
});
module.exports = __toCommonJS(cell_exports);
var import_clsx = __toESM(require("clsx"));
var import_composite = require("../composite/index.cjs");
var import_tooltip = __toESM(require("../tooltip/index.cjs"));
var import_visually_hidden = require("../visually-hidden/index.cjs");
var import_utils = require("./utils.cjs");

// packages/components/src/alignment-matrix-control/style.module.scss
var css = `/**
 * SCSS Variables.
 *
 * Please use variables from this sheet to ensure consistency across the UI.
 * Don't add to this sheet unless you're pretty sure the value will be reused in many places.
 * For example, don't add rules to this sheet that affect block visuals. It's purely for UI.
 */
/**
 * Colors
 */
/**
 * Fonts & basic variables.
 */
/**
 * Typography
 */
/**
 * Grid System.
 * https://make.wordpress.org/design/2019/10/31/proposal-a-consistent-spacing-system-for-wordpress/
 */
/**
 * Radius scale.
 */
/**
 * Elevation scale.
 */
/**
 * Dimensions.
 */
/**
 * Mobile specific styles
 */
/**
 * Editor styles.
 */
/**
 * Block & Editor UI.
 */
/**
 * Block paddings.
 */
/**
 * React Native specific.
 * These variables do not appear to be used anywhere else.
 */
.style-module__grid-container__AuKvW {
  direction: ltr;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  box-sizing: border-box;
  aspect-ratio: 1;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
}

.style-module__grid-row__xCH47 {
  grid-column: 1/-1;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
}

.style-module__cell__Ovdp9 {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  appearance: none;
  border: none;
  outline: none;
}

.style-module__point__N-8S1 {
  display: block;
  contain: strict;
  box-sizing: border-box;
  width: 6px;
  aspect-ratio: 1;
  margin: auto;
  color: var(--wp-components-color-gray-400, #ccc);
  /* Use border instead of background color so that the point shows
  in Windows High Contrast Mode */
  border: 3px solid currentColor;
  /* Highlight active item */
}
.style-module__cell__Ovdp9[data-active-item] .style-module__point__N-8S1 {
  color: var(--wp-components-color-foreground, #1e1e1e);
  transform: scale(1.6666666667);
}
.style-module__point__N-8S1 {
  /* Hover styles for non-active items */
}
.style-module__cell__Ovdp9:not([data-active-item]):hover .style-module__point__N-8S1 {
  color: var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
}
.style-module__point__N-8S1 {
  /* Show an outline only when interacting with keyboard */
}
.style-module__cell__Ovdp9[data-focus-visible] .style-module__point__N-8S1 {
  outline: 1px solid var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  outline-offset: 1px;
}
@media not (prefers-reduced-motion) {
  .style-module__point__N-8S1 {
    transition-property: color, transform;
    transition-duration: 120ms;
    transition-timing-function: linear;
  }
}`;
document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(css));
var style_module_default = {
  "grid-container": "style-module__grid-container__AuKvW",
  "grid-row": "style-module__grid-row__xCH47",
  "cell": "style-module__cell__Ovdp9",
  "point": "style-module__point__N-8S1"
};

// packages/components/src/alignment-matrix-control/cell.tsx
var import_jsx_runtime = require("react/jsx-runtime");
function Cell({
  id,
  value,
  ...props
}) {
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_tooltip.default, {
    text: import_utils.ALIGNMENT_LABEL[value],
    children: /* @__PURE__ */ (0, import_jsx_runtime.jsxs)(import_composite.Composite.Item, {
      id,
      render: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("span", {
        ...props,
        className: (0, import_clsx.default)(style_module_default.cell, props.className),
        role: "gridcell"
      }),
      children: [/* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_visually_hidden.VisuallyHidden, {
        children: value
      }), /* @__PURE__ */ (0, import_jsx_runtime.jsx)("span", {
        className: style_module_default.point,
        role: "presentation"
      })]
    })
  });
}
