// @wordpress/commands/build/store/actions.cjs
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// packages/commands/src/store/actions.js
var actions_exports = {};
__export(actions_exports, {
  close: () => close,
  open: () => open,
  registerCommand: () => registerCommand,
  registerCommandLoader: () => registerCommandLoader,
  unregisterCommand: () => unregisterCommand,
  unregisterCommandLoader: () => unregisterCommandLoader
});
module.exports = __toCommonJS(actions_exports);
function registerCommand(config) {
  return {
    type: "REGISTER_COMMAND",
    ...config
  };
}
function unregisterCommand(name) {
  return {
    type: "UNREGISTER_COMMAND",
    name
  };
}
function registerCommandLoader(config) {
  return {
    type: "REGISTER_COMMAND_LOADER",
    ...config
  };
}
function unregisterCommandLoader(name) {
  return {
    type: "UNREGISTER_COMMAND_LOADER",
    name
  };
}
function open() {
  return {
    type: "OPEN"
  };
}
function close() {
  return {
    type: "CLOSE"
  };
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  close,
  open,
  registerCommand,
  registerCommandLoader,
  unregisterCommand,
  unregisterCommandLoader
});
