// @wordpress/preferences/build/components/preference-toggle-menu-item/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// packages/preferences/src/components/preference-toggle-menu-item/index.tsx
var preference_toggle_menu_item_exports = {};
__export(preference_toggle_menu_item_exports, {
  default: () => PreferenceToggleMenuItem
});
module.exports = __toCommonJS(preference_toggle_menu_item_exports);
var import_data = require("@wordpress/data");
var import_components = require("@wordpress/components");
var import_i18n = require("@wordpress/i18n");
var import_icons = require("@wordpress/icons");
var import_a11y = require("@wordpress/a11y");
var import_store = require("../../store/index.cjs");
var import_jsx_runtime = require("react/jsx-runtime");
function PreferenceToggleMenuItem({
  scope,
  name,
  label,
  info,
  messageActivated,
  messageDeactivated,
  shortcut,
  handleToggling = true,
  onToggle = () => null,
  disabled = false
}) {
  const isActive = (0, import_data.useSelect)(
    (select) => !!select(import_store.store).get(scope, name),
    [scope, name]
  );
  const { toggle } = (0, import_data.useDispatch)(import_store.store);
  const speakMessage = () => {
    if (isActive) {
      const message = messageDeactivated || (0, import_i18n.sprintf)(
        /* translators: %s: preference name, e.g. 'Fullscreen mode' */
        (0, import_i18n.__)("Preference deactivated - %s"),
        label
      );
      (0, import_a11y.speak)(message);
    } else {
      const message = messageActivated || (0, import_i18n.sprintf)(
        /* translators: %s: preference name, e.g. 'Fullscreen mode' */
        (0, import_i18n.__)("Preference activated - %s"),
        label
      );
      (0, import_a11y.speak)(message);
    }
  };
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(
    import_components.MenuItem,
    {
      icon: isActive ? import_icons.check : null,
      isSelected: isActive,
      onClick: () => {
        onToggle();
        if (handleToggling) {
          toggle(scope, name);
        }
        speakMessage();
      },
      role: "menuitemcheckbox",
      info,
      shortcut,
      disabled,
      children: label
    }
  );
}
