// @wordpress/block-editor/build/components/html-element-control/messages.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.htmlElementMessages = void 0;
var _i18n = require("@wordpress/i18n");
/**
 * WordPress dependencies
 */

/**
 * Messages providing helpful descriptions for HTML elements.
 */
const htmlElementMessages = exports.htmlElementMessages = {
  a: (0, _i18n.__)('The <a> element should be used for links that navigate to a different page or to a different section within the same page.'),
  article: (0, _i18n.__)('The <article> element should represent a self-contained, syndicatable portion of the document.'),
  aside: (0, _i18n.__)("The <aside> element should represent a portion of a document whose content is only indirectly related to the document's main content."),
  button: (0, _i18n.__)('The <button> element should be used for interactive controls that perform an action on the current page, such as opening a modal or toggling content visibility.'),
  div: (0, _i18n.__)('The <div> element should only be used if the block is a design element with no semantic meaning.'),
  footer: (0, _i18n.__)('The <footer> element should represent a footer for its nearest sectioning element (e.g.: <section>, <article>, <main> etc.).'),
  header: (0, _i18n.__)('The <header> element should represent introductory content, typically a group of introductory or navigational aids.'),
  main: (0, _i18n.__)('The <main> element should be used for the primary content of your document only.'),
  nav: (0, _i18n.__)('The <nav> element should be used to identify groups of links that are intended to be used for website or page content navigation.'),
  section: (0, _i18n.__)("The <section> element should represent a standalone portion of the document that can't be better represented by another element.")
};
