// @wordpress/preferences/build/private-apis.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// packages/preferences/src/private-apis.ts
var private_apis_exports = {};
__export(private_apis_exports, {
  privateApis: () => privateApis
});
module.exports = __toCommonJS(private_apis_exports);
var import_preference_base_option = __toESM(require("./components/preference-base-option/index.cjs"));
var import_preference_toggle_control = __toESM(require("./components/preference-toggle-control/index.cjs"));
var import_preferences_modal = __toESM(require("./components/preferences-modal/index.cjs"));
var import_preferences_modal_section = __toESM(require("./components/preferences-modal-section/index.cjs"));
var import_preferences_modal_tabs = __toESM(require("./components/preferences-modal-tabs/index.cjs"));
var import_lock_unlock = require("./lock-unlock.cjs");
var privateApis = {};
(0, import_lock_unlock.lock)(privateApis, {
  PreferenceBaseOption: import_preference_base_option.default,
  PreferenceToggleControl: import_preference_toggle_control.default,
  PreferencesModal: import_preferences_modal.default,
  PreferencesModalSection: import_preferences_modal_section.default,
  PreferencesModalTabs: import_preferences_modal_tabs.default
});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  privateApis
});
