// @wordpress/components/build-module/angle-picker-control/angle-circle.mjs
// packages/components/src/angle-picker-control/angle-circle.tsx
import clsx from "clsx";
import { useEffect, useRef } from "@wordpress/element";
import { __experimentalUseDragging as useDragging } from "@wordpress/compose";

// packages/components/src/angle-picker-control/style.module.scss
var css = `/**
 * SCSS Variables.
 *
 * Please use variables from this sheet to ensure consistency across the UI.
 * Don't add to this sheet unless you're pretty sure the value will be reused in many places.
 * For example, don't add rules to this sheet that affect block visuals. It's purely for UI.
 */
/**
 * Colors
 */
/**
 * Fonts & basic variables.
 */
/**
 * Typography
 */
/**
 * Grid System.
 * https://make.wordpress.org/design/2019/10/31/proposal-a-consistent-spacing-system-for-wordpress/
 */
/**
 * Radius scale.
 */
/**
 * Elevation scale.
 */
/**
 * Dimensions.
 */
/**
 * Mobile specific styles
 */
/**
 * Editor styles.
 */
/**
 * Block & Editor UI.
 */
/**
 * Block paddings.
 */
/**
 * React Native specific.
 * These variables do not appear to be used anywhere else.
 */
.style-module__circle-root__j1e41 {
  border-radius: 50%;
  border: 1px solid var(--wp-components-color-gray-600, #949494);
  box-sizing: border-box;
  cursor: grab;
  height: 32px;
  overflow: hidden;
  width: 32px;
}
.style-module__circle-root__j1e41:active {
  cursor: grabbing;
}

.style-module__circle-indicator-wrapper__sbrph {
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 100%;
}
.style-module__circle-indicator-wrapper__sbrph:focus-visible {
  outline: none;
}

.style-module__circle-indicator__bS-go {
  background: var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  border-radius: 50%;
  box-sizing: border-box;
  display: block;
  left: 50%;
  top: 4px;
  transform: translateX(-50%);
  position: absolute;
  width: 6px;
  height: 6px;
}`;
document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(css));
var style_module_default = {
  "circle-root": "style-module__circle-root__j1e41",
  "circle-indicator-wrapper": "style-module__circle-indicator-wrapper__sbrph",
  "circle-indicator": "style-module__circle-indicator__bS-go"
};

// packages/components/src/angle-picker-control/angle-circle.tsx
import { jsx as _jsx } from "react/jsx-runtime";
function AngleCircle({
  value,
  onChange,
  className,
  ...props
}) {
  const angleCircleRef = useRef(null);
  const angleCircleCenterRef = useRef();
  const previousCursorValueRef = useRef();
  const setAngleCircleCenter = () => {
    if (angleCircleRef.current === null) {
      return;
    }
    const rect = angleCircleRef.current.getBoundingClientRect();
    angleCircleCenterRef.current = {
      x: rect.x + rect.width / 2,
      y: rect.y + rect.height / 2
    };
  };
  const changeAngleToPosition = (event) => {
    if (event === void 0) {
      return;
    }
    event.preventDefault();
    event.target?.focus();
    if (angleCircleCenterRef.current !== void 0 && onChange !== void 0) {
      const {
        x: centerX,
        y: centerY
      } = angleCircleCenterRef.current;
      onChange(getAngle(centerX, centerY, event.clientX, event.clientY));
    }
  };
  const {
    startDrag,
    isDragging
  } = useDragging({
    onDragStart: (event) => {
      setAngleCircleCenter();
      changeAngleToPosition(event);
    },
    onDragMove: changeAngleToPosition,
    onDragEnd: changeAngleToPosition
  });
  useEffect(() => {
    if (isDragging) {
      if (previousCursorValueRef.current === void 0) {
        previousCursorValueRef.current = document.body.style.cursor;
      }
      document.body.style.cursor = "grabbing";
    } else {
      document.body.style.cursor = previousCursorValueRef.current || "";
      previousCursorValueRef.current = void 0;
    }
  }, [isDragging]);
  return (
    // eslint-disable-next-line jsx-a11y/no-static-element-interactions
    /* @__PURE__ */ _jsx("div", {
      ref: angleCircleRef,
      onMouseDown: startDrag,
      className: clsx("components-angle-picker-control__angle-circle", style_module_default["circle-root"], className),
      ...props,
      children: /* @__PURE__ */ _jsx("div", {
        style: value ? {
          transform: `rotate(${value}deg)`
        } : void 0,
        className: clsx("components-angle-picker-control__angle-circle-indicator-wrapper", style_module_default["circle-indicator-wrapper"]),
        tabIndex: -1,
        children: /* @__PURE__ */ _jsx("div", {
          className: clsx("components-angle-picker-control__angle-circle-indicator", style_module_default["circle-indicator"])
        })
      })
    })
  );
}
function getAngle(centerX, centerY, pointX, pointY) {
  const y = pointY - centerY;
  const x = pointX - centerX;
  const angleInRadians = Math.atan2(y, x);
  const angleInDeg = Math.round(angleInRadians * (180 / Math.PI)) + 90;
  if (angleInDeg < 0) {
    return 360 + angleInDeg;
  }
  return angleInDeg;
}
var angle_circle_default = AngleCircle;
export {
  angle_circle_default as default
};
