// @wordpress/components/build/alignment-matrix-control/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// packages/components/src/alignment-matrix-control/index.tsx
var alignment_matrix_control_exports = {};
__export(alignment_matrix_control_exports, {
  AlignmentMatrixControl: () => AlignmentMatrixControl,
  default: () => alignment_matrix_control_default
});
module.exports = __toCommonJS(alignment_matrix_control_exports);
var import_clsx = __toESM(require("clsx"));
var import_i18n = require("@wordpress/i18n");
var import_compose = require("@wordpress/compose");
var import_element = require("@wordpress/element");
var import_cell = __toESM(require("./cell.cjs"));
var import_composite = require("../composite/index.cjs");
var import_icon = __toESM(require("./icon.cjs"));
var import_utils = require("./utils.cjs");

// packages/components/src/alignment-matrix-control/style.module.scss
var css = `/**
 * SCSS Variables.
 *
 * Please use variables from this sheet to ensure consistency across the UI.
 * Don't add to this sheet unless you're pretty sure the value will be reused in many places.
 * For example, don't add rules to this sheet that affect block visuals. It's purely for UI.
 */
/**
 * Colors
 */
/**
 * Fonts & basic variables.
 */
/**
 * Typography
 */
/**
 * Grid System.
 * https://make.wordpress.org/design/2019/10/31/proposal-a-consistent-spacing-system-for-wordpress/
 */
/**
 * Radius scale.
 */
/**
 * Elevation scale.
 */
/**
 * Dimensions.
 */
/**
 * Mobile specific styles
 */
/**
 * Editor styles.
 */
/**
 * Block & Editor UI.
 */
/**
 * Block paddings.
 */
/**
 * React Native specific.
 * These variables do not appear to be used anywhere else.
 */
.style-module__grid-container__AuKvW {
  direction: ltr;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr);
  box-sizing: border-box;
  aspect-ratio: 1;
  border: 1px solid transparent;
  border-radius: 4px;
  outline: none;
  cursor: pointer;
}

.style-module__grid-row__xCH47 {
  grid-column: 1/-1;
  box-sizing: border-box;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
}

.style-module__cell__Ovdp9 {
  position: relative;
  display: flex;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  appearance: none;
  border: none;
  outline: none;
}

.style-module__point__N-8S1 {
  display: block;
  contain: strict;
  box-sizing: border-box;
  width: 6px;
  aspect-ratio: 1;
  margin: auto;
  color: var(--wp-components-color-gray-400, #ccc);
  /* Use border instead of background color so that the point shows
  in Windows High Contrast Mode */
  border: 3px solid currentColor;
  /* Highlight active item */
}
.style-module__cell__Ovdp9[data-active-item] .style-module__point__N-8S1 {
  color: var(--wp-components-color-foreground, #1e1e1e);
  transform: scale(1.6666666667);
}
.style-module__point__N-8S1 {
  /* Hover styles for non-active items */
}
.style-module__cell__Ovdp9:not([data-active-item]):hover .style-module__point__N-8S1 {
  color: var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
}
.style-module__point__N-8S1 {
  /* Show an outline only when interacting with keyboard */
}
.style-module__cell__Ovdp9[data-focus-visible] .style-module__point__N-8S1 {
  outline: 1px solid var(--wp-components-color-accent, var(--wp-admin-theme-color, #3858e9));
  outline-offset: 1px;
}
@media not (prefers-reduced-motion) {
  .style-module__point__N-8S1 {
    transition-property: color, transform;
    transition-duration: 120ms;
    transition-timing-function: linear;
  }
}`;
document.head.appendChild(document.createElement("style")).appendChild(document.createTextNode(css));
var style_module_default = {
  "grid-container": "style-module__grid-container__AuKvW",
  "grid-row": "style-module__grid-row__xCH47",
  "cell": "style-module__cell__Ovdp9",
  "point": "style-module__point__N-8S1"
};

// packages/components/src/alignment-matrix-control/index.tsx
var import_jsx_runtime = require("react/jsx-runtime");
function UnforwardedAlignmentMatrixControl({
  className,
  id,
  label = (0, import_i18n.__)("Alignment Matrix Control"),
  defaultValue = "center center",
  value,
  onChange,
  width = 92,
  ...props
}) {
  const baseId = (0, import_compose.useInstanceId)(UnforwardedAlignmentMatrixControl, "alignment-matrix-control", id);
  const setActiveId = (0, import_element.useCallback)((nextActiveId) => {
    const nextValue = (0, import_utils.getItemValue)(baseId, nextActiveId);
    if (nextValue) {
      onChange?.(nextValue);
    }
  }, [baseId, onChange]);
  const classes = (0, import_clsx.default)("component-alignment-matrix-control", style_module_default["grid-container"], className);
  return /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_composite.Composite, {
    defaultActiveId: (0, import_utils.getItemId)(baseId, defaultValue),
    activeId: (0, import_utils.getItemId)(baseId, value),
    setActiveId,
    rtl: (0, import_i18n.isRTL)(),
    render: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", {
      ...props,
      className: classes,
      "aria-label": label,
      id: baseId,
      role: "grid",
      style: {
        width: `${width}px`
      }
    }),
    children: import_utils.GRID.map((cells, index) => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_composite.Composite.Row, {
      render: /* @__PURE__ */ (0, import_jsx_runtime.jsx)("div", {
        className: style_module_default["grid-row"],
        role: "row"
      }),
      children: cells.map((cell) => /* @__PURE__ */ (0, import_jsx_runtime.jsx)(import_cell.default, {
        id: (0, import_utils.getItemId)(baseId, cell),
        value: cell
      }, cell))
    }, index))
  });
}
var AlignmentMatrixControl = Object.assign(UnforwardedAlignmentMatrixControl, {
  /**
   * Render an alignment matrix as an icon.
   *
   * ```jsx
   * import { AlignmentMatrixControl } from '@wordpress/components';
   *
   * <Icon icon={<AlignmentMatrixControl.Icon value="top left" />} />
   * ```
   */
  Icon: Object.assign(import_icon.default, {
    displayName: "AlignmentMatrixControl.Icon"
  })
});
var alignment_matrix_control_default = AlignmentMatrixControl;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AlignmentMatrixControl
});
