// @wordpress/wordcount/build/defaultSettings.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// packages/wordcount/src/defaultSettings.ts
var defaultSettings_exports = {};
__export(defaultSettings_exports, {
  defaultSettings: () => defaultSettings
});
module.exports = __toCommonJS(defaultSettings_exports);
var defaultSettings = {
  HTMLRegExp: /<\/?[a-z][^>]*?>/gi,
  HTMLcommentRegExp: /<!--[\s\S]*?-->/g,
  spaceRegExp: /&nbsp;|&#160;/gi,
  HTMLEntityRegExp: /&\S+?;/g,
  // \u2014 = em-dash.
  connectorRegExp: /--|\u2014/g,
  // Characters to be removed from input text.
  removeRegExp: new RegExp(
    [
      "[",
      // Basic Latin (extract)
      "!-/:-@[-`{-~",
      // Latin-1 Supplement (extract)
      "\x80-\xBF\xD7\xF7",
      /*
       * The following range consists of:
       * General Punctuation
       * Superscripts and Subscripts
       * Currency Symbols
       * Combining Diacritical Marks for Symbols
       * Letterlike Symbols
       * Number Forms
       * Arrows
       * Mathematical Operators
       * Miscellaneous Technical
       * Control Pictures
       * Optical Character Recognition
       * Enclosed Alphanumerics
       * Box Drawing
       * Block Elements
       * Geometric Shapes
       * Miscellaneous Symbols
       * Dingbats
       * Miscellaneous Mathematical Symbols-A
       * Supplemental Arrows-A
       * Braille Patterns
       * Supplemental Arrows-B
       * Miscellaneous Mathematical Symbols-B
       * Supplemental Mathematical Operators
       * Miscellaneous Symbols and Arrows
       */
      "\u2000-\u2BFF",
      // Supplemental Punctuation.
      "\u2E00-\u2E7F",
      "]"
    ].join(""),
    "g"
  ),
  // Remove UTF-16 surrogate points, see https://en.wikipedia.org/wiki/UTF-16#U.2BD800_to_U.2BDFFF
  astralRegExp: /[\uD800-\uDBFF][\uDC00-\uDFFF]/g,
  wordsRegExp: /\S\s+/g,
  characters_excluding_spacesRegExp: /\S/g,
  /*
   * Match anything that is not a formatting character, excluding:
   * \f = form feed
   * \n = new line
   * \r = carriage return
   * \t = tab
   * \v = vertical tab
   * \u00AD = soft hyphen
   * \u2028 = line separator
   * \u2029 = paragraph separator
   */
  characters_including_spacesRegExp: /[^\f\n\r\t\v\u00AD\u2028\u2029]/g,
  l10n: {
    type: "words"
  }
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  defaultSettings
});
